src/grt/src/fastroute/src/maze3D.cpp

```diff
@@ -982,7 +983,8 @@ void FastRouteCore::mazeRouteMSMDOrder3D(int expand,
               && directions_3D[curL][curY][curX] != Direction::East) {
             const float tmp = d1_3D[curL][curY][curX] + 1;
             if (h_edges_3D_[curL][curY][curX - 1].usage
-                < h_edges_3D_[curL][curY][curX - 1].cap) {
+                < h_edges_3D_[curL][curY][curX - 1].cap &&
+                net->minLayer <= curL && curL <= net->maxLayer) {
```

Comment:
How was the layer previously checked?

Reply:
Previously we had different runs for different layer restrictions, so this function wasn't looking at the layers out of the range.